tests/end2end/fixtures/quteprocess.py

```diff
@@ -113,7 +114,18 @@ def formatted_str(self, colorized=True):
         r.module = self.module
         r.funcName = self.function
 
-        formatter = log.ColoredFormatter(log.EXTENDED_FMT, log.DATEFMT, '{',
+        format_str = log.EXTENDED_FMT
+        # Mark expected errors with (Expected) so it's less confusing for tests
+        # which expect errors but fail due to other errors.
+        if self.expected and self.loglevel >= log.LOG_LEVELS['ERROR']:
+            new_color = '{' + log.LOG_COLORS['DEBUG'] + '}'
+            format_str = format_str.replace('{log_color}', new_color)
+            format_str = re.sub(r'{levelname:(\d*)}',
+                                # Leave away the padding because (Expected) is
+                                # longer anyway.
+                                r'{levelname} (Expected)', format_str)
```

Comment:
I don't know why, but I think "Expected" (instead of "expected") looks weird :wink: